core/src/main/scala/filodb.core/downsample/DownsampleCardinalityManager.scala

```diff
@@ -0,0 +1,170 @@
+package filodb.core.downsample
+
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.StrictLogging
+
+import filodb.core.DatasetRef
+import filodb.core.memstore.PartKeyLuceneIndex
+import filodb.core.memstore.ratelimit.{CardinalityTracker, QuotaSource, RocksDbCardinalityStore}
+import filodb.core.metadata.PartitionSchema
+import filodb.core.store.StoreConfig
+import filodb.memory.format.UnsafeUtils
+
+/**
+ * Manages the calculation and storage of cardinality count in a shard of downsample cluster
+ */
+class DownsampleCardinalityManager(rawDatasetRef: DatasetRef,
+                                   shardNum: Int,
+                                   shardKeyLen: Int,
+                                   partKeyIndex: PartKeyLuceneIndex,
+                                   partSchema: PartitionSchema,
+                                   filodbConfig: Config,
+                                   downsampleStoreConfig: StoreConfig,
+                                   quotaSource: QuotaSource) extends StrictLogging {
+
+  var cardTracker: CardinalityTracker = initCardTracker()
+
+  // number of shards per node - used to stagger/distribute the cardinality count calculation of different shards
+  // to different times of day
+  private val numShardsPerNode: Int = getNumShardsPerNodeFromConfig(rawDatasetRef.dataset, filodbConfig)
+
+  /**
+   * Helper method to get the shards per node in the downsample filodb cluster. We look for the two specific
+   * config that is `min-num-nodes` ( the minimum number of nodes/pods which has to be in a cluster ) and
+   * `num-shards` (the total number of shards). num-shards/min-num-nodes gives us the required value.
+   *
+   * NOTE: This configs can be part of `dataset-configs` or `inline-dataset-configs`. Also making sure, we look
+   * for the dataset, which was used to create this instance of DownsampleTimeSeriesShard
+   *
+   * @return shards present per node
+   */
+  def getNumShardsPerNodeFromConfig(datasetToSearch: String, filoConfig: Config): Int = {
```

Comment:
Is this equivalent? https://github.com/filodb/FiloDB/blob/654b7b03c9f0dea8ec81185230deb84970be162d/coordinator/src/main/scala/filodb.coordinator/ShardAssignmentStrategy.scala#L238

Reply:
(not sure yet whether-or-not we have access to a `DatasetResourceSpec`)

Reply:
Yes, it is equivalent. Thanks @alextheimer for the find.  Let me see if we have access to it ^^